trusted_shuffler/trusted_shuffler/src/lib.rs

```diff
@@ -33,45 +33,77 @@ use std::{
 use tokio::{sync::oneshot, time::Duration};
 
 #[derive(Debug, PartialEq)]
-pub struct TrustedShufflerRequest {
+pub struct EncryptedRequest {
+    // TODO(#3282) Unify `body` and `data` from Request and Response.
     pub body: Vec<u8>,
     pub headers: hyper::HeaderMap,
     pub uri: Uri,
 }
 
-impl TrustedShufflerRequest {
-    // We only compare the body, and we need revisit this and decide if this
-    // is how we want to sort our requests.
-    fn compare_body(&self, r2: &TrustedShufflerRequest) -> Ordering {
+// TODO(#3281): Add decryption.
+impl EncryptedRequest {
+    fn decrypt(self) -> PlaintextRequest {
```

Comment:
I don't think an encrypted request will have a method to decrypt itself, anyways I guess this will be changed in the future.